### DISCUSSION TITLE
Best practices for multiple teams/services/environments

### DISCUSSION THREAD

**Discussion by chriselion** [healthchecks/healthchecks#1001] [Q&A]
How does everybody structure their projects and checks to support multiple independent teams and services? For example, we might have something like

```
TeamA
    Service1
        staging - alerts go to #team-a-alerts slack channel
        production - alerts go to Service1 PagerDuty service
    Service2
        staging - alerts go to #team-a-alerts slack channel
        production - alerts go to Service2 PagerDuty service
TeamB
    Service3
        staging - alerts go to #team-b-alerts slack channel
        production - alerts go to Service3 PagerDuty service
```

I'm thinking about having one Healthchecks Project per service, each of which has 1 PagerDuty and 1 Slack integration, then the checks for that service can select the integration based on the environment. That seems a bit tedious to manage the various API keys, though (since I want to use terraform to manage checks). 

It's also a bit of "click-ops" to add a new team or service. There's no way to API for adding integrations, is there?

Is there a better way to go about this? I wouldn't mind having everything in one Project, except that adding a new Integration adds itself to all existing checks.

**Comment by chriselion** [comment]
Another approach I'm considering is a Project per environment; that would make the API key management a little easier. That would also mean a potentially large number of integrations per project, and I'm worried that unselecting a new integration from all existing checks would get tedious for users (and there's a potential for mis-clicks).

**Comment by cuu508** [comment]
I'm a team of one, so unfortunately can't share much first-hand experience.

Another way to structure the checks you could consider – use separate Healthchecks.io accounts for prod and staging:

* Account prod@yourcompany.com
   * Project: Service1 
     * team members: TeamA
     * integrations: Service1 PagerDuty
   * Project: Service2
     * team members: TeamA
     * integrations: Service2 PagerDuty
   * Project: Service3 
     * team members: TeamB
     * integrations: Service3 PagerDuty
* Account staging@yourcompany.com
   * Project: Service1 
     * team members: TeamA
     * integrations: #team-a-alerts
   * Project: Service2
     * team members: TeamA
     * integrations: #team-a-alerts
   * Project: Service3 
     * team members: TeamB
     * integrations: #team-b-alerts

This way, prod and staging would be fully separated. You can give developers access to projects in the staging account with no risk of affecting prod. (but you can still invite the same developer in prod and staging projects at the same time, they will not need to juggle multiple accounts). You can also use staging API keys and not risk accidentally nuking production checks. Each project would have a single integration. You would still have many API keys if there are many services.

Another quick note: there's a way to un-assign all checks from an integration in a single operation, but it is a little hidden: go to the Integrations screen, and click on a cell in the "Assigned Checks" column. You will see a dialog where you can select/unselect all:

![image](https://github.com/healthchecks/healthchecks/assets/661859/04931277-e3f7-487c-ad1d-da0a6328d5ca)

**Comment by chriselion** [comment]
Thanks @cuu508 - I'll keep the multi-account option in mind too.

Regarding adding integrations: I agree with all the suggestions raised [here](https://github.com/healthchecks/healthchecks/issues/903) - ideally I'd like a per-project setting to control how new integrations interact with existing checks, but a checkbox when adding a new integration would be fine too.
